src/renderer/utils.ts

```diff
@@ -22,8 +22,8 @@ export function buildOpdsBrowserRoute(
 
 // Encode without padding
 // And apply RFC 4648
-export function encodeB64(data: any) {
-    let encoded = btoa(data);
+export function encodeB64(data: string) {
+    let encoded = btoa(unescape(encodeURIComponent(data)));
```

Comment:
I need to review this carefully, based on the actual usage of the `encodeB64()` function.
I speak from extensive experience -- in developing the `r2-xxx-js` packages in particular, where Base64 and URL/URI encoding is critical, and error-prone -- working with the subtleties of `unescape`/`escape`, `decodeURIComponent`/`encodeURIComponent` and also `decodeURI`/`encodeURI`).

I will come back to this later (sorry cannot do it right now).

Reply:
I need this to works : https://stackoverflow.com/questions/23223718/failed-to-execute-btoa-on-window-the-string-to-be-encoded-contains-characte

Reply:
FYI, related issue: https://github.com/readium/readium-desktop/issues/486
(let's handle this in a separate Pull Request)

Reply:
This Pull Request (481) is about fixing / harmonizing the HTTP request utility functions.

I suggest that the Base64-related code changes (which impact the OPDS React router routes) should be removed, and treated in a separate Pull Request. Could you please file an issue to explain what the problem is exactly (reproducible bug), or alternatively just create a Pull Request that explains the issue itself. Thank you.

Reply:
Ok, but my PR become obsolete without this

Reply:
How create a new issue, if the bug is reproducible only with a specific accept-language on request for me ?

Reply:
I understand the original logical coupling of HTTP request <-> Base64 encoding/decoding, due to how the Base64 bug surfaced in your tests. It is now also apparent that there should be two separate well-scoped Pull Requests of single responsibility. We must be able to test and validate the Base64 code alterations outside of the context of the refactoring of the HTTP API, which - as you have observed - revealed several subtle side effects.

In another PR (I will file issues later) we must properly handle `httpGet()` promise rejections (`try`+`catch`) and `finally` on the client side, so that error recovery from the `request` `error` parameter (e.g. request timeout), or from totally unexpected exceptions (which also reject the promise) can be used to manage the UI state (e.g. progress bar). I saw that the Redux-Saga `yield call` already handles rejection correctly, but others do not (resulting in "unhandled promise rejections").

Let's have a separate "clean" PR (based on `develop`, using the harmonized/centralized HTTP utility function) to address the Base64 encoding/decoding (as you probably saw, there is already a separate issue about broken handling of `+` characters). This will be easier to test and rollback if necessary.

Reply:
ok , let's do it

But explain me the particularity of encodeB64 and decodeB64 : it's used only for the opdsBrowserRoute. the drift is limited

Reply:
Sure, but for example in other `r2-xxx-js` packages, we use an `encodeURIComponent()` implementation that conforms to RFC3986 (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent ) ...so, better isolate these types of code changes.

Reply:
New issue: https://github.com/readium/readium-desktop/issues/522